requirements/requirements-dev.txt

```diff
@@ -11,6 +11,7 @@ Sphinx==8.0.2
 sphinxcontrib-bibtex==2.6.2
 sphinx-copybutton==0.5.2
 sphinx-autodoc-typehints==2.0.0
+sphix-tags
```

Comment:
check for dependency conflicts

Reply:
I think we might need to scrap this plan or wait for `sphinx-tags` to be compatible with python 3.12. Looking through their release and test workflows, it seems this extension is compatible with python 3.11 only. 

https://github.com/melissawm/sphinx-tags/blob/2cbb9eed1f6762e6a20ef378c6d88fa1a833b65d/.github/workflows/release.yml#L19

One workaround to this is to use `pip install sphinx-tags` after `make install` because `pip` somehow misses the conflicts in this case.